src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1570,6 +1570,21 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
                                     op2->SetUnusedValue();
                                 }
 
+                                // Since we have a double negation, it's possible that gtNext
+                                // is op1 or user. If it is op1, then it's also possible the
+                                // subsequent gtNext is user. We need to make sure to skip both
+                                // in such a scenario since we're removing them.
+
+                                if (nextNode == op1)
+                                {
+                                    nextNode = nextNode->gtNext;
+                                }
+
+                                if (nextNode == user)
+                                {
+                                    nextNode = nextNode->gtNext;
+                                }
```

Comment:
This isn't my favorite fix, but given where we are in .NET 9 it's probably the safest one.

The `TernaryLogic` lowering being done by looking at the `user` is already a little different compared to other lowering logic, but it's how it got introduced several months back. More ideally we'd be handling this closer to how general containment gets handled, which is looking at the operands of a node instead.

That is, rather than saying "I am a `BitwiseOper`, is the node that consumes me also a `BitwiseOper`, we should probably be saying `I am a `BitwiseOper`, are either of my operands also `BitwiseOper`.

Such handling would avoid issues like this and also allow us to more easily take into account whether the `ternarylogic` usage is "profitable" (since it may not be if the two individual nodes could have been contained, as an example).

Reply:
I notably wasn't able to get a simplified repro for this either, even running through the Antigen trimmer the smallest repro it found was 2100 lines of code.

Morph and other phases normally recognize/handle the double NOT much earlier and so its not typically found in practice.

Reply:
I think I can understand the logic and the reason why you have to skip `op1` and `user` for gtNext (I wonder if it's possible to just mark them as UnusedValue to keep them in LIR chain and continue LIR walk normally, but probably not). 
And I agree that'd be much more natural to do this op while visiting the root XOR instead.